server/src/main/java/com/linecorp/centraldogma/server/internal/api/CredentialServiceV1.java

```diff
@@ -111,9 +118,15 @@ public CompletableFuture<Credential> getCredentialById(User loginUser,
     @ConsumesJson
     @StatusCode(201)
     @Post("/projects/{projectName}/credentials")
-    @RequiresRepositoryRole(value = RepositoryRole.WRITE, repository = Project.REPO_META)
+    @RequiresProjectRole(ProjectRole.OWNER)
     public CompletableFuture<PushResultDto> createCredential(@Param String projectName,
-                                                             Credential credential, Author author, User user) {
+                                                             CreateCredentialRequest request,
+                                                             Author author, User user) {
+        final String credentialId = request.credentialId();
+        final String credentialName = credentialName(projectName, credentialId);
+        // Ignore the credential name in the request and set the name with the format of
+        // "projects/{project}/credentials/{credentialId}".
+        final Credential credential = request.credential().withName(credentialName);
```

Comment:
As the credential name in `Credential` is not used, should we make the field nullable so that the client doesn't have to fill it?

Reply:
@ikhoon The `name` is nullable when a credential is created:
https://github.com/line/centraldogma/blob/aec299376aa2baefcbba198baa608fff6b71ae51/server/src/main/java/com/linecorp/centraldogma/server/internal/credential/AccessTokenCredential.java#L42|
Do you mean the front-end side?

Reply:
I was confused. I thought the client should fill `credential.name` in the request body.
From my understanding, the value is optional so clients could omit the value.

Reply:
That's correct. 👍